plugin-maven/src/test/java/com/diffplug/gradle/spotless/MavenRunner.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.gradle.spotless;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.assertj.core.api.Assertions;
+
+import com.diffplug.common.base.Throwables;
+
+/**
+ * Harness for running a maven build, same idea as the
+ * [GradleRunner from the gradle testkit](https://docs.gradle.org/current/javadoc/org/gradle/testkit/runner/GradleRunner.html).
+ */
+public class MavenRunner {
+	public static MavenRunner create() {
+		return new MavenRunner();
+	}
+
+	private MavenRunner() {}
+
+	private File projectDir;
+	private String[] args;
+
+	public MavenRunner withProjectDir(File projectDir) {
+		this.projectDir = Objects.requireNonNull(projectDir);
+		return this;
+	}
+
+	public MavenRunner withArguments(String... args) {
+		this.args = Objects.requireNonNull(args);
+		return this;
+	}
+
+	private Result run() throws IOException, InterruptedException {
+		Objects.requireNonNull(projectDir, "Need to call withProjectDir() first");
+		Objects.requireNonNull(args, "Need to call withArguments() first");
+		// run maven with the given args in the given directory
+		List<String> cmds = getPlatformCmds("mvn " + Arrays.stream(args).collect(Collectors.joining(" ")));
+		ProcessBuilder builder = new ProcessBuilder(cmds);
+		builder.directory(projectDir);
+		Process process = builder.start();
+		// slurp and return the stdout, stderr, and exitValue
+		Slurper output = new Slurper(process.getInputStream());
+		Slurper error = new Slurper(process.getErrorStream());
```

Comment:
@nedtwigg If we expect `MavenRunner` to be instantiated many times, I think we should consider using a shared `ExecutorService`, like an `Executors.newFixedThreadPool(Runtime.getRuntime().getAvailableProcessors() - 1)`. Otherwise, this whole commit looks pretty reasonable to me. :)

Reply:
To clarify, I'm proposing we use an `ExecutorService` instead of using `Thread`s directly through the `Slurper` class.

Reply:
Roger.  There will only be 10's of integration tests, so I don't think there's much of a performance hit for starting 2x threads per.

Reply:
Okay, that sounds reasonable to me.